src/containerapp/azext_containerapp/custom.py

```diff
@@ -648,7 +648,7 @@ def create_containerapp(cmd,
         container_def["volumeMounts"] = [volume_mount_def]
         template_def["volumes"] = [volume_def]
 
-    if revision_suffix is not None:
+    if revision_suffix is not None and not is_registry_msi_system(registry_identity):
```

Comment:
Will this break the scenario that only sets revision suffix?

Reply:
```
def is_registry_msi_system(identity):
    if identity is None:
        return False
    return identity.lower() == "system"
```

no, as shown code, for the scenario that only sets revision suffix, is_registry_msi_system return false, so the value will be set.